jkube-kit/build/api/src/test/java/org/eclipse/jkube/kit/build/api/assembly/AssemblyConfigurationUtilsTest.java

```diff
@@ -56,11 +62,14 @@ void getAssemblyConfigurationOrCreateDefault_withNoConfiguration_shouldReturnDef
   @Test
   void getAssemblyConfigurationOrCreateDefault_withConfiguration_shouldReturnConfiguration() {
     // Given
-    BuildConfiguration mockedBuildConfiguration = mock(BuildConfiguration.class);
-    final AssemblyConfiguration configuration = AssemblyConfiguration.builder().user("OtherUser").name("ImageName").build();
-    when(mockedBuildConfiguration.getAssembly()).thenReturn(configuration);
+    BuildConfiguration buildConfig = BuildConfiguration.builder()
```

Comment:
why are you creating a local variable of the same name as class level `buildConfig` here?

Reply:
Actually test _getAssemblyConfigurationOrCreateDefault_withConfiguration_shouldReturnConfiguration_ is fails when I define buildConfig here like this:
`
buildConfig.toBuilder()
            .assembly(AssemblyConfiguration.builder()
                    .user("OtherUser")
                    .name("ImageName")
                    .build())
            .build();
`
after declaring it as above, I'm getting assembly configurations as null.
That's why I have redeclared it again.

Reply:
Does it work if you reassign `buildConfig` variable to new value?

Reply:
Could you please replace it by directly reassigning `buildConfig`?

Reply:
Sorry, I was confused with terms.
It not works as expected when I reassign the new value to `buildConfig`.
When I reassign `buildConfig` variable to new value I'm getting `buildConfig.getAssembly()` as `null`.
and when I declare it as  
```
BuildConfiguration buildConfig = BuildConfiguration.builder()
            .assembly(AssemblyConfiguration.builder()
                    .user("OtherUser")
                    .name("ImageName")
                    .build())
            .build();
```

it works as expected.
I don't know why this things happens.

Reply:
I'm asking you to modify it like this:

```suggestion
    buildConfig = buildConfig.toBuilder()
```

Reply:
Sure, Thanks! It works.